Assets/HoloToolkit/Input/Scripts/InputSources/MotionControllerInputSource.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.XR.WSA.Input;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Motion Controller Input source for processing input from motion controllers.  For each of these you would
+    /// want to specify handedness or not.  Based on handedness indicates which controller this source would track
+    /// </summary>
+    public class MotionControllerInputSource : BaseInputSource
```

Comment:
Pretty sure we already have a class that does exactly this, but in a more generalized way.

@keveleigh ?

Reply:
I didn't see anywhere that had motion controller input.  All the other scripts duplicated source delegates rather than using InputSource.

Reply:
We have https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Unity_2017.2.0/Assets/HoloToolkit/Input/Scripts/InputSources/InteractionSourceInputSource.cs for motion controller input, though this does a couple things not present there.

Reply:
I also want to make sure the Motion controllers and Xbox Controllers play nicely.

Reply:
Ah, so InteractionSourceInputSource (that's a lot of source!) is for both HoloLens gesture and motion controller then.  The only things I'd add is the ability to set handedness to filter for on controllers and accessors for linear and angular velocity.

thoughts?

thanks,
-pat

Reply:
I'd say make those awesome changes to the `InteractionSources` class.  I like that the class is more generalized and cross platform friendly.  Makes things less complex.

Reply:
Agreed! (Also agreed on "that's a lot of source!" haha! Definitely open to rename suggestions 😄). I added handedness as a variable in the source data, but I don't believe there's any way to actually pass that through yet. I think the two best ways would be to add a TryGetHandedness or to add it to each of the event data classes. Thoughts?

Accessors for velocity would be great as well!

Reply:
I think `InteractionInputSource` is fine lol.

Reply:
I just updated it to have accessors for velocities in this fork.  I'll look to merge changes you've already made in as well.

thanks,
-pat